orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -178,27 +178,65 @@ class TaskController {
     }
   }
 
+/**
+ * Retrieves an ad-hoc collection of executions based on a number of user-supplied parameters. Either executionIds or
+ * pipelineConfigIds must be supplied in order to return any results
+ * @param pipelineConfigIds A comma-separated list of pipeline config ids; ignored if executionIds is also specified
+ * @param executionIds A comma-separated list of execution ids; if specified, all other parameters, except expand,
+ * will be ignored
+ * @param limit (optional) Number of most recent executions to retrieve per pipeline config; defaults to 1, ignored if
+ * executionIds is specified
+ * @param statuses (optional) Execution statuses to filter results by; defaults to all, ignored if executionIds is
+ * specified
+ * @param expand (optional) Expands each execution object in the resulting list. If this value is missing,
+ * it is defaulted to true.
+ * @return
+ */
+  @PostFilter("hasPermission(filterObject.application, 'APPLICATION', 'READ')")
   @RequestMapping(value = "/pipelines", method = RequestMethod.GET)
-  List<Execution> listLatestPipelines(
-    @RequestParam(value = "pipelineConfigIds") String pipelineConfigIds,
+  List<Execution> listSubsetOfPipelines(
+    @RequestParam(value = "pipelineConfigIds", required = false) String pipelineConfigIds,
+    @RequestParam(value = "executionIds", required = false) String executionIds,
     @RequestParam(value = "limit", required = false) Integer limit,
-    @RequestParam(value = "statuses", required = false) String statuses) {
+    @RequestParam(value = "statuses", required = false) String statuses,
+    @RequestParam(value = "expand", defaultValue = "true") boolean expand) {
     statuses = statuses ?: ExecutionStatus.values()*.toString().join(",")
     limit = limit ?: 1
-    def executionCriteria = new ExecutionRepository.ExecutionCriteria(
+    ExecutionRepository.ExecutionCriteria executionCriteria = new ExecutionRepository.ExecutionCriteria(
       limit: limit,
       statuses: (statuses.split(",") as Collection)
     )
 
-    def ids = pipelineConfigIds.split(',')
+    if (!pipelineConfigIds && !executionIds) {
+      return []
+    }
 
-    def allPipelines = rx.Observable.merge(ids.collect {
-      executionRepository.retrievePipelinesForPipelineConfigId(it, executionCriteria)
-    }).subscribeOn(Schedulers.io()).toList().toBlocking().single().sort(startTimeOrId)
+    if (executionIds) {
+      List<String> ids = executionIds.split(',')
 
-    return filterPipelinesByHistoryCutoff(allPipelines, limit)
-  }
+      List<Execution> executions = rx.Observable.from(ids.collect {
+        executionRepository.retrieve(PIPELINE, it)
+      }).subscribeOn(Schedulers.io()).toList().toBlocking().single()
 
+      if (!expand) {
+        unexpandPipelineExecutions(executions)
+      }
+
+      return executions
+    } else {
```

Comment:
doesn't seem like you need the actual `else`?

Reply:
True, I'll remove.